### PR TITLE
Completion for available topic in kafka file

### LINKED ISSUE #150: Completion for available topic in kafka file
Completion for topic property to show available topic of the selected cluster

### PR DESCRIPTION
**Pull request by angelozerr**
Completion for available topic in kafka file

Fixes #150

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

**Comment by angelozerr** [comment]
It starts working but I need to clean my code and write tests.

When completion is applied for topic (property name or value) it connect to the cluster to get the topics (just the first time). I wonder if this connection must be done only with kafka explorer?

**Comment by angelozerr** [comment]
The topics completion is working only if you try to open completion after `topic:` but it doesn't work if you try to open `topic:` completion item and apply it (topics as choice). 

I did this choice because when cluster is not available it breaks the full property names completion support (because the topics values for choice cannot be done). Is it possible to see this topics value as choice in an another issue?

File: src/kafka-file/kafkaFileClient.ts [hunk 1]
```diff
@@ -5,13 +5,15 @@ import { ClusterSettings } from "../settings/clusters";
...
-import { ConsumerLaunchStateProvider, getLanguageService, LanguageService, ProducerLaunchStateProvider, SelectedClusterProvider } from "./languageservice/kafkaFileLanguageService";
+import { ConsumerLaunchStateProvider, getLanguageService, LanguageService, ProducerLaunchStateProvider, SelectedClusterProvider, TopicDetail, TopicProvider } from "./languageservice/kafkaFileLanguageService";
 import { runSafeAsync } from "./utils/runner";
+import { KafkaExplorer, TopicItem } from "../explorer";
 
 export function startLanguageClient(
     clusterSettings: ClusterSettings,
     producerCollection: ProducerCollection,
     consumerCollection: ConsumerCollection,
+    explorer: KafkaExplorer,
```
Comment (fbricon) [inline src/kafka-file/kafkaFileClient.ts:16]: don't inject the KafkaExplorer, which is too UI oriented, but a KafkaModelProvider instance instead

**Review by fbricon — CHANGES_REQUESTED** [review]

File: src/kafka-file/kafkaFileClient.ts (hunk 1 above)
Comment (angelozerr) [reply src/kafka-file/kafkaFileClient.ts]: Good catch! Fixed.

**Comment by fbricon** [comment]
Can you add en entry to the changelog?

**Comment by angelozerr** [comment]
> Can you add en entry to the changelog?

done

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,7 @@ All notable changes to `Tools for Apache Kafka®` are documented in this file.
 - Declare key/value formats for PRODUCER in kafka file. See [#113](https://github.com/jlandersen/vscode-kafka/issues/113).
 - Completion support for property names and values of CONSUMER and PRODUCER blocks. See [#146](https://github.com/jlandersen/vscode-kafka/issues/146).
 - Completion support for fakerJS PRODUCER key and value. See [#152](https://github.com/jlandersen/vscode-kafka/issues/152).
+- Completion support for available topic of CONSUMER and PRODUCER blocks. See [#150](https://github.com/jlandersen/vscode-kafka/issues/150).
```
Comment (fbricon) [inline CHANGELOG.md:11]: ... available topics for ....

Comment (angelozerr) [reply CHANGELOG.md]: fixed

**Comment by fbricon** [comment]
Thanks @angelozerr!

**Pull request merged**